@here/olp-sdk-dataservice-read/lib/utils/getTile.ts

```diff
@@ -138,13 +144,19 @@ export async function getTile(
         quadTreeIndex.subQuads.length
     ) {
         if (fetchOptions !== FetchOptions.OnlineOnly) {
-            cache.put(partitionsRequest, catalogHrn.toString(), layerId, [
-                {
-                    partition: partitionId,
-                    dataHandle: quadTreeIndex.subQuads[0].dataHandle,
-                    version: quadTreeIndex.subQuads[0].version
-                }
-            ]);
+            const subQuadsToPartitionsList = quadTreeIndex.subQuads.map(
```

Comment:
you need to also cache the parentQuads.

Reply:
I'm caching on 161 line

Reply:
Sorry, missed that. Ignore this comment.